vignettes/example_analysis.Rmd

```diff
@@ -1,104 +1,334 @@
+
+```{r thedata}
+library(surveyTools)
+dat <- surveyTools::EQ5D5L_surveys |>
+  dplyr::mutate(surveyID = factor(surveyID)) |>
+  # Split population into age groups
+  dplyr::mutate(AgeGroup = cut(age, c(20, 40, 60, Inf), right = F))
+
+dat |>
+  dplyr::select(respondentID, surveyID, mobility, self_care,
+                usual, pain, anxiety, AgeGroup, sex) |>
+  dplyr::arrange(surveyID, respondentID) |>
+  head() |>
+  knitr::kable(caption = "Data excerpt")
+```
+
+Next we calculate the utility values based on the EQ-5D survey.
+
+```{r loaddata}
+input_dat <- as_eq5d5l(
+    dat,
+    mobility = "mobility",
+    self_care = "self_care",
+    usual = "usual",
+    pain = "pain",
+    anxiety = "anxiety",
+    respondentID = "respondentID",
+    surveyID = "surveyID",
+    time_index = "time_index",
+    vas = "vas"
+) |>
+  add_utility(dat, type = "DSU", country = "UK", age = "age", sex = "sex")
+```
+
+We will be using a mixed effect model to fit to the data. The model is defined as `value ~ (1 + acute | respondentID) + surveyID + sex + AgeGroup + sex:AgeGroup`. This means that we assume each respondent to have a random effect as well as a random interaction between the first survey after symptoms (acute). Other explanatory variables are the survey, sex and age group (and the interaction between sex and age group). This model was chosen based on our knowledge of the data. For your own dataset it is recommended to choose your own model. For example, by exploring multiple models and choosing the best model (using standard model comparison methods).
+
+To fit our mixed effect model we use the `lme4` package. While the utility value is truncated at 1, it has been shown that assuming a normal distribution and performing a non parametric bootstrap is a valid simplification [@pull_analysis_2010]. We will explore a more complex approach later in this vignette.
+
+```{r lmerModel, fig.cap = "Comparing the modelled utility values with the utility values in the model. Utility values were calculated from a EQ5D5L survey using the DSU method."}
+# Label the acute period of the disease
+input_dat |>
+  dplyr::mutate(acute = surveyID == "survey02") -> dat
+
+# Define the model
+model <- .value ~ (1 + acute | respondentID) +
+  surveyID + sex + AgeGroup + sex:AgeGroup
+lme4::lmer(model, data = dat) -> fit2
+
+# Compare predictions to the actual value
+dat |>
+  dplyr::mutate(Model = predict(fit2)) |>
+  dplyr::rename(Actual = .value) |>
+  tidyr::gather(type, value, Actual, Model) -> plot_dat
```

Comment:
use pivot_longer for user facing vignette as best to match current recommendations.